modules/social_features/social_event/src/Form/EnrollRequestDeclineForm.php

```diff
@@ -104,9 +135,17 @@ public function buildForm(array $form, FormStateInterface $form_state) {
 
     // Load the user profile to format a nice name.
     if (!empty($this->eventEnrollment)) {
-      /** @var \Drupal\profile\Entity\Profile $user_profile */
-      $user_profile = Profile::load($this->eventEnrollment->getAccount());
-      $this->fullName = $user_profile->field_profile_first_name->value . ' ' . $user_profile->field_profile_last_name->value;
+      /** @var \Drupal\profile\Entity\ProfileInterface[] $user_profiles */
+      $user_profiles = $this->entityTypeManager
+        ->getStorage('profile')
+        ->loadByProperties(['uid' => $this->eventEnrollment->getAccount()]);
+
+      foreach ($user_profiles as $user_profile) {
```

Comment:
We know that there will always be only one account attached to the expected enrolment, we should avoid unnecessary foreach structures, you can use reset() and get the first array element:

```
      $user_profiles = $this->entityTypeManager
        ->getStorage('profile')
        ->loadByProperties(['uid' => $this->eventEnrollment->getAccount()]);

      $user_profile = reset($user_profiles);
      $this->fullName = $this->socialProfileNameService->getProfileName($user_profile);
```

Reply:
That was my first approach as well. But I looked at some other modules e.g ldap_profile and researched what to use. It seems that a foreach should be better performance-wise than a function call.